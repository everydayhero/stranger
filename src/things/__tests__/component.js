/* global describe it expect */
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import comp from '../component.js'

describe('comp', () => {
  it('should remove invalid html tags from React element based components', () => {
    const StyledComp = comp(() => ({
      color: 'red'
    }))('div')
    const component = renderer.create(
      <StyledComp notAValidProp='Noooo!' />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should not remove invalid html tags from native React Components', () => {
    const StyledComp = comp(() => ({
      color: 'red',
      backgroundColor: 'blue'
    }))('div')
    const StyledCompInehrit = comp(() => ({
      color: 'white'
    }))(StyledComp)
    const component = shallow(
      <StyledCompInehrit notAValidProp='Noooo!' />
    )
    expect(component.props()).toMatchSnapshot()
  })
})
