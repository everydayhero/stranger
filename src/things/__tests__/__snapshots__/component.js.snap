exports[`comp should not remove invalid html tags from native React Components 1`] = `
Object {
  "className": "cxsync-2906296101",
  "notAValidProp": "Noooo!",
  "styles": Object {
    "color": "white",
  },
}
`;

exports[`comp should remove invalid html tags from React element based components 1`] = `
<div
  className="cxsync-722373509" />
`;
